### PR TITLE
fix: propagate text-align to column groups

### LINKED ISSUE #1673: ColumnTextAlign is not applied to all header rows
Vaadin 14.0.2

### Description
Parameter ColumnTextAlign is applied only to last header row (first footer row). 

### Expected outcome
Text inside all headers (footers) cells should be aligned accordingly.

### Code snippet
```
final Grid<String> g = new Grid<>();
final Column<?> col = g.addColumn(String::toString).setHeader("Header 1").setTextAlign(ColumnTextAlign.END);

g.prependHeaderRow().getCell(col).setText("Header 0");
g.appendHeaderRow().getCell(col).setText("Header 2");
g.appendHeaderRow().getCell(col).setText("Header 3");

g.appendFooterRow().getCell(col).setText("Footer 0");
g.appendFooterRow().getCell(col).setText("Footer 1");
g.appendFooterRow().getCell(col).setText("Footer 2");
```

### Actual outcome
Capture

### PR DESCRIPTION
**Pull request by DiegoCardoso**
## Description

Propagate the `text-align` attribute to parent column groups. It will only apply the same `text-align` value if the parent group does not span to multiple columns.

Fixes #1673

## Type of change

- [X] Bugfix
- [ ] Feature

### CODE REVIEW

File: vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/ColumnGroupHelpers.java [hunk 1]
```diff
@@ -105,4 +105,20 @@ private static ColumnGroup wrapSingleColumn(AbstractColumn<?> column,
...
     }
 
+    /**
+     * Copy the text align value from each column on the source list to its
+     * correspondent column on the destination list
+     *
+     * @param source
+     *            list of columns the text align values are copied from
+     * @param destination
+     *            list of columns the text align values are copied tox
```
Comment (tomivirkki) [inline vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/ColumnGroupHelpers.java:115]: typo `tox`

File: vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/ColumnBase.java [hunk 2]
```diff
@@ -120,6 +120,16 @@ default boolean isFrozenToEnd() {
     default T setTextAlign(ColumnTextAlign textAlign) {
         getElement().setProperty("textAlign",
                 textAlign == null ? null : textAlign.getPropertyValue());
+
+        // Propagate text align to parent columns
+        if (getElement().getParent() != null) {
+            getElement().getParent().getComponent()
+                    .filter(parent -> parent instanceof AbstractColumn<?>
+                            && parent.getChildren().count() == 1)
```
Comment (tomivirkki) [inline vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/ColumnBase.java:128]: This condition is untested

**Review by tomivirkki — APPROVED** [review]
Added a couple of comments, but the fix works great as is.

File: vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/ColumnBase.java (hunk 2 above)
Comment (DiegoCardoso) [reply vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/ColumnBase.java:128]: Added a test that sets text-align to a column child of a joined group.

File: vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/ColumnGroupHelpers.java (hunk 1 above)
Comment (DiegoCardoso) [reply vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/ColumnGroupHelpers.java]: Fixed.

**Pull request merged**